code/modules/projectiles/gun.dm

```diff
@@ -424,6 +424,27 @@
 				damage_mult = 3
 	P.damage *= damage_mult
 
+/obj/item/weapon/gun/proc/get_dispersion_range(mob/user)
+	var/dt = world.time - last_shot_time
+	var/dt_movement = world.time - user.last_movement
+	var/recoil_range = recoil / ergonomics
+	var/accuracy_range = accuracy
+
+	if(dt > firemodes[sel_mode].burst_delay)
+		recoil_range /= sqrt(dt) * 1.5
+		if((recoil_range -= sqrt(dt) * 1.5) < 0)
```

Comment:
Do: if((recoil_range - sqrt(dt) * 1.5) < 0)
If you do -= you change your car inside the if function which is bad in most cases

Reply:
Ok. But it was not a bug. I was just trying to avoid code duplication.

Reply:
Alright! That's why I said in most cases, if it's intentional it's fine for merge